modApiServer/src/org/aion/api/server/pb/ApiAion0.java

```diff
@@ -1284,6 +1340,149 @@ public ApiAion0(IAionChain ac) {
                 return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_function_exception_VALUE);
             }
         }
+        case Message.Funcs.f_getBlockSqlByRange_VALUE: {
+            if (service != Message.Servs.s_admin_VALUE) {
+                return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_service_call_VALUE);
+            }
+
+            byte[] data = parseMsgReq(request, msgHash);
+            Message.req_getBlockSqlByRange req;
+
+            try {
+                if (LOG.isDebugEnabled()) LOG.debug("BlockSqlByRange: start");
+
+                req = Message.req_getBlockSqlByRange.parseFrom(data);
+                long latestBlkNum = this.getBestBlock().getNumber();
+
+                Long _blkStart = req.getBlkNumberStart();
+                Long _blkEnd = req.getBlkNumberEnd();
+
+                // no null check here
+
+                long blkStart = -1;
+                long blkEnd = -1;
+
+                if (_blkStart < 0)
+                    blkStart = 0;
+                else
+                    blkStart = _blkStart;
+
+                if (_blkEnd > latestBlkNum)
+                    blkEnd = latestBlkNum;
+                else
+                    blkEnd = _blkEnd;
+
+                if (blkEnd < blkStart)
+                    return ApiUtil.toReturnHeader(getApiVersion(), Message.Retcode.r_fail_function_arguments_VALUE);
+
+                // truncate the thing
+                if (blkEnd - blkStart > 1000) {
+                    blkStart = blkEnd - 1000 + 1;
+
+                    if (blkStart < 0) blkStart = 0;
+                }
+
+                if (LOG.isDebugEnabled()) LOG.debug("BlockSqlByRange: range " + blkStart + "-" + blkEnd);
+
+                Long lastBlockTimestamp = null;
+                long listLength = blkEnd - blkStart + 1;
```

Comment:
Just to be clear here, this implies our domain is [blkStart, blkEnd]

Reply:
yes, domain is [blkStart, blkEnd]

Reply:
LGTM